src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2597,6 +2602,7 @@ private BoundExpression BindRangeExpression(RangeExpressionSyntax node, BindingD
             CheckFeatureAvailability(node, MessageID.IDS_FeatureRangeOperator, diagnostics);
 
             TypeSymbol rangeType = GetWellKnownType(WellKnownType.System_Range, diagnostics, node);
+            ReportDiagnosticsIfObsolete(diagnostics, rangeType, node, hasBaseReceiver: false);
```

Comment:
I looked around for other references to `WellKnownType.System_Range/System_Index` to see if more scenarios have a similar issue.
The ones related to implicit index/range indexers seem fine.
But a few more places need closer look (likely are failing to report obsolete diagnostic):
- slice patterns (see `BindSlicePattern`),
- inline array index (see `tryImplicitConversionToInlineArrayIndex` and `bindInlineArrayElementAccess`). It looks like the scenario may require an implicit user-defined conversion from the operand type to `System.Index` or `System.Range`
- `BindRangeExpressionOperand` uses `WellKnownType.System_Index`
Could you add a test for these scenarios and either fix them or file a tracking issue?